### PR TITLE
Set m.room.canonical_alias on room creation.

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/handlers/room.py [hunk 1]
```diff
@@ -271,6 +272,15 @@ def create(etype, content, **kwargs):
 
             returned_events.append(power_levels_event)
 
+        if room_alias:
+            if (EventTypes.CanonicalAlias, '') not in initial_state:
```
Comment (NegativeMjark) [inline synapse/handlers/room.py]: Probably don't need two levels of ``if`` block here:

    if room_alias and (EventTypes.CanonicalAlias, '') not in initial_state:

**Comment by NegativeMjark** [comment]
Otherwise LGTM.

**Comment by erikjohnston** [comment]
Done

**Comment by NegativeMjark** [comment]
LGTM

**Pull request merged**